### PR TITLE
[Bugfix] Fondy and additional bugs fixed #2921

### PR DESCRIPTION
**Pull request by Edw1x**
Fixed Fondy bug
Fixed 3 step

### CODE REVIEW

File: src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.html [hunk 1]
```diff
@@ -117,6 +117,9 @@ <h3 class="bottom_comment_payment">{{ 'order-form.payment' | translate }}</h3>
     <button class="secondary-global-button btn cansel-button btn-main" (click)="cancelUBS()">
       {{ 'order-details.cancel' | translate }}
     </button>
-    <button (click)="redirectToOrder()" type="submit" class="primary-global-button btn">{{ 'submit-order.to-order' | translate }}</button>
+    <button (click)="redirectToOrder()" type="submit" class="primary-global-button btn">
+      <span *ngIf="loadingAnim" class="spinner-border spinner-border-sm" role="status" aria-hidden="true"> </span>
```
Comment (JimHolman1199) [inline src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.html:121]: why do we need an empty span ?

File: src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts [hunk 2]
```diff
@@ -63,6 +71,19 @@ export class UBSSubmitOrderComponent extends FormBaseComponent implements OnInit
...
 
   redirectToOrder() {
-    document.location.href = this.shareFormService.orderUrl;
+    this.loadingAnim = true;
+    this.orderService
+      .getOrderUrl()
+      .pipe(takeUntil(this.destroy))
+      .subscribe(
+        (fondyUrl) => {
+          this.shareFormService.orderUrl = fondyUrl.toString();
```
Comment (JimHolman1199) [inline src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts:80]: will this not be string by default ?

File: src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.html (hunk 1 above)
Comment (Edw1x) [reply src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.html:121]: Forgot to remove it.

File: src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts (hunk 2 above)
Comment (Edw1x) [reply src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts:80]: I will investigate this.

Comment (Edw1x) [reply src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts:80]: Yea, it's a string, but I understood why there was a .toString() method.
image

File: src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.html (hunk 1 above)
Comment (JimHolman1199) [reply src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.html:121]: let's use spinner from shared folder

Comment (Edw1x) [reply src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.html:121]: We need there a spinner for a button, in shared folder we have spinner for page loader
My solution with span:
image

**Review by JimHolman1199 — APPROVED** [review]

**Pull request merged**